### PR TITLE
[#4763] prevent navigation back to transaction

### LINKED ISSUE #4760: Back button in "Transaction sent" should navigate back to the chat or wallet
### Description

*Type*: Bug

*Summary*: If transaction was sent, pressing the device back button should return user to wallet or chat screen depending on the place transaction was initiated. It should basically work the same as pressing "Got it". Instead, the back button leads back to "Send transaction" screen.

#### Expected behavior
Pressing back button in "Transaction sent" should navigate to either wallet or chat depending on the place transaction was initiated.

#### Actual behavior
The back button leads back to "Send transaction" screen.

### Reproduction

- Open Status on Android
- Open 1-1 chat
- Send ("/send") transaction and press device back button in "Transaction sent" screen 

### Additional Information
* Status version: nightly, June 14
* Operating System: Android

### LINKED ISSUE #4763: Read-only fields Amount, Asset and Recipient can be modified if return back after transaction was sent in chat
### Description

*Type*: Bug

*Summary*: Read-only fields Amount, Asset and Recipient can be modified if return back after transaction was sent in chat

#### Expected behavior
Fields can't be modified or there is no Send transaction screen if tap back, so user needs to make new /send 

#### Actual behavior
after user sends 1 transaction in 1:1 chat and taps on 'Got it' she can return back to Sign transaction screen with device back button. On this screen read-only fields Amount, Asset and Recipient can be modified and new transaction can be sent. Unfortunately, 1:1 chat will show initial value and not the one used in transaction. 

### Reproduction
from 00:17 https://app.testfairy.com/projects/4803622-status/builds/8494723/sessions/4393959670/?accessToken=JUDd5PA90-3eZLP5RwANpW0twfk

- Open Status
- Open 1:1 chat
- send 1 transaction, tap Got it
- tap device back button. As a result, Send transaction screen is shown
- Try to edit field Amount and send transaction
- Check send message in 1:1 chat and transaction history for the actual amount sent

### Additional Information
* Status version: nightly build june 15 and PR build https://github.com/status-im/status-react/pull/4761#issuecomment-397216921
* Operating System: Android

### PR DESCRIPTION
**Pull request by rasom**
fix #4763 
fix #4760

status: ready

### CODE REVIEW

**Review by jeluard — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

**Comment by lukaszfryc** [comment]
@rasom there's a bug. The chat does not show "/sent" message after transaction is sent. 

Other findings:
1. Back button works fine in all scenarios (it fixes https://github.com/status-im/status-react/issues/4760)
2. This PR should probably fix https://github.com/status-im/status-react/issues/4759

**Comment by lukaszfryc** [comment]
@rasom the issue is still there

File: src/status_im/ui/screens/wallet/send/events.cljs [hunk 1]
```diff
@@ -411,6 +411,6 @@
  (fn [{:keys [db]} [_ chat-id]]
    {:dispatch       (condp = (second (:navigation-stack db))
                       :wallet-send-transaction [:navigate-to-clean :wallet]
-                      :wallet-send-transaction-chat [:execute-stored-command-and-return-to-chat chat-id]
+                      :chat [:execute-stored-command-and-return-to-chat chat-id]
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/send/events.cljs]: This looks like a very fragile piece of code, @goranjovic does it look fine for you?

**Comment by rasom** [comment]
@lukaszfryc when a new build will be ready please check #4760 again

**Comment by lukaszfryc** [comment]
@rasom still the same. No "/sent" command in 1-1 chat when I go through the usual flow without even pressing back button. 

Does it work on your side?

**Comment by rasom** [comment]
@lukaszfryc the last build definitely works on my side

**Comment by lukaszfryc** [comment]
@rasom the "/send" message is visible when I click "Got it" and return back to the chat 👍 
But, if I go back instead of clicking "Got it", the message is not present. So, this issue is still not fixed: https://github.com/status-im/status-react/issues/4759

**Comment by rasom** [comment]
@lukaszfryc objective here is #4763, if this one is fixed we are good to merge and work #4759 in the separate PR

**Comment by lukaszfryc** [comment]
@rasom it makes sense. So only #4760 and #4763 are fixed in this PR.

**Pull request merged**